lib/rpmsg/rpmsg_virtio.c

```diff
@@ -689,11 +690,11 @@ int rpmsg_init_vdev_with_config(struct rpmsg_virtio_device *rvdev,
 		 * Since device is RPMSG Remote so we need to manage the
 		 * shared buffers. Create shared memory pool to handle buffers.
 		 */
+		rvdev->shpool = config->split_shpool ? shpool + 1 : shpool;
```

Comment:
If config->split_shpool is set to a true value other than 1, this will give bad values for rvdev->shpool. I suggest making the value explicitly 1 or 2.

Reply:
@edmooring, could you re-check? for me it Ok as split_shpool is a boolean, but perhaps I missed something

Reply:
@arnopo, you are correct. I didn't look at enough of the context.